docs/development.md

```diff
@@ -116,3 +116,28 @@ You'll need to create a test repository and install your app by clicking the "In
 
 1. Always run `$ npm install` and restart the server if `package.json` has changed.
 1. To turn on verbose logging, start server by running: `$ LOG_LEVEL=trace npm start`
+
+## Alternate way of running a probot app
+
+If you take a look to the `npm start` script, this is what it runs: `probot run ./index.js`. This is nice, but you sometimes need more control on how your Node.js application is executed. For example if you want to use custom V8 flags, `ts-node`, etc. you need more flexibility. In those cases there's a simple way of executing your probot application programmatically:
+
+```js
+// main.js
+const { Probot } = require('probot')
+const app = require('./index.js')
+
+// pass a probot app as a function
+Probot.run(app)
+```
+
+or
+
+```js
+// main.js
+const { Probot } = require('probot')
+
+// or pass the same arguments you would pass to the `probot` command
+Probot.run(process.argv)
```

Comment:
Do you see a use case for this? I think I’d only support the `Probot.run(app)` API. Even if `Probot.run(process.argv)` works due to the way it is implemented, I’d not document it, unless we can think of use cases.

Reply:
One use case is if you just want to create an index.js file that you can invoke with your custom v8 flags or with `ts-node` and you want to keep the default argv behavior.

This was Brandon's ide, btw.

Reply:
when you have a custom `index.js` file which invokes the `Probot.run()` function, then you would start your app with `node index.js`. Would you pass your v8 flags to that?

I’d prefer to remove the bit and add it once a legit use case comes up for which we can’t offer a better alternative

Reply:
Yes, like for example

> node --experimental-worker index.js

Reply:
Also if you want to run `ts-node` and use the same parameters than the probot binary.

Related to this https://github.com/probot/probot/issues/743

Reply:
when I do

```
node --experimental-worker index.js
```

Then I already passed the `--experimental-worker` flag to the process, why would we need to pass it to `Probot.run(process.argv)` again?

And wouldn't `ts-node  index.js` would just work I think?

I think I’m missing something here

Reply:
As I said here https://github.com/probot/probot/pull/751#discussion_r257599365

> and you want to keep the default argv behavior

meaning, you want to keep using probot's flags.

So with `Probot.run(process.argv)` you can keep using probot's flags **plus** v8 flags and/or run your app with another binary (e.g. `ts-node`).

Reply:
I think this creates a conflict since we also check if `appFn` is an array [here](https://github.com/probot/probot/pull/751/files#diff-f41e9d04a45c83f3b6f6e630f10117feR75), so someone might want to pass `Probot.run(['appFn1', 'appFn2'])` and expect it to run both modules, but this would parse that as arguments. What if we took a second optional object for the arguments, so you could do this?

```js
Probot.run(['appFn1', 'appFn2'], {
  args?: ['run', 'file.js']
})
```

That way we keep `Probot.run(appFn)` or `Probot.run(['appFn1', 'appFn2'])` syntax, which could default to `['run', 'index.js']`. Still pondering this, so not sure it's the best solution at the moment, but that seems to make sense to me.

Reply:
Sorry for the back-and-forth :)

> and you want to keep the default argv behavior

I’d argue that we don’t need it. Either I use the probot binary with some defaults and CLI arguments that we think cover most of people’s use cases, or they use the JavaScript API and pass what ever they want, for the remaining cases. I don’t think we need a mixture of both. I think using `Probot.run()` implies and advanced use case, if people use that then they should know what they are doing. I’m still not convinced that we need to support `Probot.run(process.argv)`. And it’s easier to add it later if it becomes a blocker for someone than removing it

Reply:
Ok. If we don't want to support `Probot.run(process.argv)` then I need to change probot-run.js as well because now probot-run depends directly on `Probot.run(process.argv)`.

Do you want me to change probot-run and Probot.run() in order to disable `Probot.run(process.argv)`?

Reply:
We can also just not document it so we can later remove it as it’s not considered a public API

Reply:
Done :)